tests/journal.rs

```diff
@@ -0,0 +1,90 @@
+use std::process::Command;
+
+use libsystemd::logging::Priority;
+use std::collections::HashMap;
+use std::ffi::OsStr;
+
+/// Read from journal.
+///
+/// `matches` are additional matches to filter journal entries.
+fn read_from_journal<I, S>(matches: I) -> Vec<HashMap<String, String>>
```

Comment:
In my experience this kind of integration tests are often flakey due to details of the underlying environment.

I'd be happier to have unit tests on `add_field_and_payload` to be honest.

Reply:
I'll add unit tests for serialization, but in the end a unit test won't actually test compatibility with journald :shrug:  

Your call though; if you'd like me to remove these tests I'll get rid of the file again.  I can also mark them as `ignored` and update the github actions file again so that these tests only run Github by default where the environment is repeatable and deterministic.

However you like :slightly_smiling_face:

Reply:
Let's keep them, and thanks for adding them in the first place!
But I have a feeling that they may cause troubles in the future and I may have to disable them. If that eventually happens, having the unit tests on the side would be good at that point (mostly for catching glaring regressions, not the full journald flow).

Reply:
I've added a couple of simple unit tests for the encoding.

Reply:
I've also added some explicit randomness to the integration tests to make extra sure that each test only sees its own journal entries; this should make the tests as reliable as it gets for integration tests.  I believe they now should only ever fail if journald isn't running.

Should I mark them as `#[ignore]` nonetheless, so that they don't get run by default?